release-calendar/src/client/models/release-event.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {EventInput} from '@fullcalendar/core/structs/event';
+import {EventSourceInput} from '@fullcalendar/core/structs/event-source';
+import {Release} from '../../types';
+
+function convertReleaseToEvent(release: Release): EventInput {
+  return {
+    title: release.name,
+    start: release.date,
+    className: release.channel,
+    extendedProps: {isRollback: release.isRollback},
+  };
+}
+
+function groupIntoChannels(events: EventInput[]): EventSourceInput[] {
+  const ltsEvents: EventInput[] = [],
+    stableEvents: EventInput[] = [],
+    percBetaEvents: EventInput[] = [],
+    percExperimentalEvents: EventInput[] = [],
+    optInBetaEvents: EventInput[] = [],
+    optInExperimentalEvents: EventInput[] = [],
+    nightlyEvents: EventInput[] = [];
+
+  events.forEach((event: EventInput) => {
+    switch (event.className) {
```

Comment:
Instead of a switch statement, what if you used a dictionary where the key was the same as `event.className`?

Reply:
I just recommitted. I was having trouble declaring the return type in the convertReleaseToEvent function so I literally defined a new object and put it into my types file. As a result, when I add each event to the dictionary I do this,
.forEach(pair => dict[pair[0]].push(pair[1])) 
on line 47 of release-event.ts. what are your thoughts?